[Zhu Wenbo] iP

mendax1234 opened the pull request.

mendax1234 committed:
Add Robonaut.java

mendax1234 committed:
Add echo method

mendax1234 committed:
Optimize the UI

mendax1234 committed:
Add add and list function

Emannuel-Tan commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+import java.lang.reflect.Array;
+import java.util.*;
```
Shouldn't Import statements be listed explicitly?

sivanshno commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+                + "|  _ \\ ___ | |__   ___  _ __   __ _ _   _| |_ \n"
+                + "| |_) / _ \\| '_ \\ / _ \\| '_ \\ / _` | | | | __|\n"
+                + "|  _ < (_) | |_) | (_) | | | | (_| | |_| | |_ \n"
+                + "|_| \\_\\___/|_.__/ \\___/|_| |_|\\__,_|\\__,_|\\__|\n";
```
I like the CLI art

Emannuel-Tan commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+        // Greetings
+        hello();
+
+        Scanner sc = new Scanner (System.in);
```
Perhaps using a better variable name instead of "sc" would better explain its use.

Emannuel-Tan commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+            case "list":
+                list(list);
+                break;
+            default:
+                add(list, option);
+                break;
```
Perhaps using a seperate case for adding tasks and using default case as a error catching/handling would be better

sivanshno commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+
+        Scanner sc = new Scanner (System.in);
+
+        // Level-2
```
Maybe it would be better to put a comment here explaining the purpose of the code block below

Emannuel-Tan commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+        ArrayList<String> list = new ArrayList<>();
+        while (!option.equals("bye")) {
+            switch (option) {
+            case "list":
```
"list" is a Magic String, Consider extracting as a constant or an enum

sivanshno commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+    public static void list(ArrayList<String> list) {
+        int i = 1;
+        System.out.println("------------------------------------------------------------");
+        for (String e : list) {
```
good job following all code conventions for the loop

sivanshno commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+    public static void main(String[] args) {
+        // Greetings
+        hello();
+
```
Is the space between line 48 and line 50 needed?

Emannuel-Tan commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+
+    // Level 1
+    public static void echo(String s) {
+        System.out.println("------------------------------------------------------------");
```
Consider extracting the dashed lines a constant and using it everytime that it is needed

Kevin88866 replied:
I think you can make "-------------------" to a constant.

Emannuel-Tan reviewed (commented):
Overall, I find your code pretty readable and extracted well with seperate functions. Keep it up!

mendax1234 committed:
Add Task class to manage Tasks

Kevin88866 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+        list.add(content);
+        System.out.println("------------------------------------------------------------");
+        System.out.println("added: " + content);
+        System.out.println("------------------------------------------------------------");
```
Can use the function echo above

Kevin88866 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+        String option = sc.nextLine();
+        ArrayList<String> list = new ArrayList<>();
+        while (!option.equals("bye")) {
+            switch (option) {
```
Can use case "list" -> list(list);

Kevin88866 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    // Level 2
+    public static void add(ArrayList<String> list, String content) {
```
This function is used to print message after adding. So the function name can be something like addMessage.

mendax1234 committed:
Add mark/unmark features

Fix the index offset by 1 problem when doing mark/unmark

Kevin88866 reviewed (commented):
Overall, it’s great, but I have a small suggestion that could make your code look simpler：）

gohjieling834 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.*;
```
Perhaps you can list the imports explicitly?

gohjieling834 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,114 @@
+
+public class Robonaut {
+    // Level 0
+    public static void hello() {
```
Maybe you can consider using verbs for the method name (e.g. printHelloMessage), so that it follows the coding standards.

gohjieling834 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,114 @@
+                + "|  _ < (_) | |_) | (_) | | | | (_| | |_| | |_ \n"
+                + "|_| \\_\\___/|_.__/ \\___/|_| |_|\\__,_|\\__,_|\\__|\n";
+        System.out.println(logo);
+        System.out.println("Hey bro! Good to see you here! I'm Robonaut, your most intelligent helper!");
```
I like your chatbot's personality!

gohjieling834 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,114 @@
+        System.out.println("------------------------------------------------------------");
+    }
+
+    public static String getStatus(boolean status) {
+        StringBuilder s = new StringBuilder();
+        s.append('[');
+        if (status) {
+            s.append('X');
+            s.append(']');
+        } else {
+            s.append(' ');
+            s.append(']');
+        }
+        return s.toString();
+    }
```
Do you still need this method, since there's already a method to getStatusIcon in the Task class?

gohjieling834 reviewed (commented):
Overall code is neat and easy to understand.

mendax1234 committed:
Optimize code standard

Add JavaDoc for every existing method
Change some coding styles according to the peer feedback

mendax1234 committed:
Optimize method name

mendax1234 committed:
Implement three types of task
Create an abstract class Task
Inherit three subclasses from Task
Put all tasks into a package
Modify the main logic in the main program -- Robonaut.java

yentheng0110 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,148 @@
+
+    public static void main(String[] args) {
+        printHelloMessage();
+
```
Perhaps you can consider removing this blank line for improved readability

yentheng0110 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,148 @@
+                String[] parts = option.substring(9).split("/by", 2);
+                printAddCommand(tasks, new Deadline(parts[0].trim(), parts[1].trim()));
+            } else if (option.startsWith("event")) {
+                String[] parts = option.substring(6).split("/from|/to");
```
The numbers 5, 9, 6 are magic numbers, perhaps you can consider abstracting them out as constants

yentheng0110 commented on src/main/java/Robonaut.java:
```diff
@@ -0,0 +1,148 @@
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        System.out.println(HORIZONTAL_LINE);
```
Perhaps you can consider abstracting out the printing of the horizontal line as a method

yentheng0110 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,59 @@
+     * @return "[X]" if done, "[ ]" otherwise
+     */
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]");
```
I like the use of ternary operator here

yentheng0110 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,59 @@
+    protected String description;
+
+    /** Whether the task is marked as done. */
+    protected boolean isDone;
```
I like how you named the boolean variable such that it sounds like a boolean (isDone)

yentheng0110 reviewed (commented):
In general, your code is clean. Minor changes were suggested

mendax1234 committed:
Improve according to TA's adivce

mendax1234 committed:
Add Auto Text UI testing support

mendax1234 committed:
Update Expected.txt

mendax1234 committed:
Add Complete Exception Handling

mendax1234 committed:
Revert "Add Complete Exception Handling"

This reverts commit b0ff5dfc4548e1c3a757657c9b208a03ed3e0b29.

mendax1234 committed:
Revert "Add Complete Exception Handling"

This reverts commit b0ff5dfc4548e1c3a757657c9b208a03ed3e0b29.

mendax1234 committed:
Add Complete Exception Handling

mendax1234 committed:
Merge branch 'branch-Level-5'

mendax1234 committed:
Add package robonaut'

mendax1234 committed:
Merge branch 'branch-A-Packages'

# Conflicts:
#	src/main/java/robonaut/Robonaut.java
#	src/main/java/robonaut/exceptions/EmptyDescriptionException.java
#	src/main/java/robonaut/exceptions/InvalidFormatException.java
#	src/main/java/robonaut/exceptions/InvalidTaskNumberException.java
#	src/main/java/robonaut/exceptions/RobonautException.java
#	src/main/java/robonaut/exceptions/UnknownCommandException.java

mendax1234 committed:
Fix package path problem

mendax1234 committed:
Restructure the whole code

mendax1234 committed:
Update automatic ut test

mendax1234 committed:
Optimize package structure

mendax1234 committed:
Add Delete Command

mendax1234 committed:
Add Storage Function

mendax1234 committed:
Update Storage function

mendax1234 committed:
Merge branch 'branch-Level-6'

mendax1234 committed:
Merge branch 'branch-Level-7'

mendax1234 committed:
Add JAR necessary files

mendax1234 committed:
Improve date format for deadline task

mendax1234 committed:
Merge branch 'branch-Level-8'

mendax1234 committed:
Add Find feature

mendax1234 committed:
Add JavaDocs for FindCommand

mendax1234 committed:
Merge branch 'branch-Level-9'

mendax1234 committed:
Finish JavaDocs for all classes

mendax1234 committed:
Add JavaDocs fo the missing classes

mendax1234 committed:
Merge branch 'branch-A-JavaDoc'

mendax1234 committed:
Add Robonaut UG

mendax1234 committed:
Update Deadline.java

mendax1234 committed:
Merge pull request #1 from mendax1234/branch-Level-8

Update Deadline.java

mendax1234 committed:
Update FindCommand

mendax1234 committed:
Merge pull request #2 from mendax1234/branch-Level-9

Update FindCommand

mendax1234 committed:
Update JavaDocs

mendax1234 committed:
Merge pull request #3 from mendax1234/branch-A-JavaDoc

Update JavaDocs

mendax1234 committed:
Improve exception handling

mendax1234 committed:
Update README.md for better clarity
